backend/core/src/listings/combined/combined-listing-query-builder.ts

```diff
@@ -0,0 +1,201 @@
+import { Listing } from "../entities/listing.entity"
+import { QueryRunner, SelectQueryBuilder } from "typeorm"
+import { Pagination } from "nestjs-typeorm-paginate"
+import { CombinedListingTransformer } from "./combined-listing-transformer"
+import { ListingFilterParams } from "../dto/listing-filter-params"
+import { addFilters } from "../../shared/query-filter"
+import { combinedListingFilterTypeToFieldMap } from "./filter-type-to-field-map"
+import { OrderByFieldsEnum } from "../types/listing-orderby-enum"
+import { OrderParam } from "../../applications/types/order-param"
+import { HttpException, HttpStatus } from "@nestjs/common"
+
+type OrderByConditionData = {
+  orderBy: string
+  orderDir: "DESC" | "ASC"
+  nulls?: "NULLS LAST" | "NULLS FIRST"
+}
+
+// REMOVE_WHEN_EXTERNAL_NOT_NEEDED
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export class CombinedListingsQueryBuilder extends SelectQueryBuilder<any> {
```

Comment:
Nit: CombinedListing and ExternalListing have many similar base fields to the literal SQL table Listing, but are much richer objects with all of the joins, etc. 

We could consider renaming CombinedListing and ExternalListing to something that represents this like `ListingApiResponse` or `RichListing` or something like that, (e.g. it may have helped with @isaac-looker 's confusion in standup the other day around "why not just save both to the same table").

Reply:
mostly tangential to this PR- it sure would be amazing to build a listings table with all sources(internal, external) and return internal or both rows from the listings or combined listings endpoint respectively. We'd need to etl from both internal listings and external  sources every few mins if there are changes. We could use this architecture even after we removed external sources, since it would also speed up queries.

Reply:
@isaac-looker The view defined by `CombinedListings` does exactly what you are proposing, but without the need to  refresh anything other than external listings.

Reply:
@lisunshiny The naming around `Combined*` was based on `includeExternal` already being taken on the service and controller and me wanting to keep naming consistent across related classes.  I disagree that naming around API responses is warranted because 1) the `Listing` class also contains rich data generated from joins and 2) the local listing data generated by queries to `combined_listings` only includes data from local related tables, not API responses.